build: boost::process detection doesn't pass include directory

`boost::process` detection fails here (FreeBSD 13.0) with:
```
configure:32183: checking for Boost Process
configure:32196: c++ -std=c++17 -c -g -O2  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -DPROVIDE_FUZZ_MAIN_FUNCTION conftest.cpp >&5
conftest.cpp:80:10: fatal error: 'boost/process.hpp' file not found
#include <boost/process.hpp>
         ^~~~~~~~~~~~~~~~~~~
1 error generated.
configure:32196: $? = 1
```
The header in question does exist in `/usr/local/include/boost/process.hpp`. However, no include path is passed?

For comparison, the preceding boost::filesystem check, which passes `-I/usr/local/include` succeeds:
```
configure:32026: checking for exit in -lboost_filesystem
configure:32051: c++ -std=c++17 -o conftest -g -O2  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -DPROVIDE_FUZZ_MAIN_FUNCTION -I/usr/local/include  -L/usr/local/lib conftest.cpp -lboost_filesystem   -lboost_system >&5
configure:32051: $? = 0
configure:32061: result: yes
```

I can confirm the same behaviour on OpenBSD 6.9:
```
configure:31594: checking for Boost Process
configure:31607: c++ -std=c++17 -c -g -O2  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -DPROVIDE_FUZZ_MAIN_FUNCTION conftest.cpp >&5
conftest.cpp:79:10: fatal error: 'boost/process.hpp' file not found
#include <boost/process.hpp>
         ^~~~~~~~~~~~~~~~~~~
1 error generated.
```

```
$ find / -name process.hpp
...
/usr/local/include/boost/process.hpp
...
```

Confirming the same behavior on my FreeBSD 13 machine; when trying to build off master:

```
checking for Boost Process... configure: error: Boost::Process is required for external signer support, but not available!
```

--- a/configure.ac
+++ b/configure.ac
@@ -1415,7 +1415,7 @@ fi
 if test x$use_boost = xyes; then
 
   dnl Check for Boost headers
-  AX_BOOST_BASE([1.58.0],[],[AC_MSG_ERROR([Boost is not available!])])
+  AX_BOOST_BASE([1.64.0],[],[AC_MSG_ERROR([Boost is not available!])])
   if test x$want_boost = xno; then
     AC_MSG_ERROR([[only libbitcoinconsensus can be built without boost]])
   fi

--- a/configure.ac
+++ b/configure.ac
@@ -1438,11 +1438,6 @@ if test x$use_boost = xyes; then
     BOOST_CPPFLAGS=SUPPRESS_WARNINGS($BOOST_CPPFLAGS)
   fi
 
-  dnl Boost 1.56 through 1.62 allow using std::atomic instead of its own atomic
-  dnl counter implementations. In 1.63 and later the std::atomic approach is default.
-  m4_pattern_allow(DBOOST_AC_USE_STD_ATOMIC) dnl otherwise it's treated like a macro
-  BOOST_CPPFLAGS="-DBOOST_SP_USE_STD_ATOMIC -DBOOST_AC_USE_STD_ATOMIC $BOOST_CPPFLAGS"
-
   BOOST_LIBS="$BOOST_LDFLAGS $BOOST_SYSTEM_LIB $BOOST_FILESYSTEM_LIB"
 fi
 

--- a/configure.ac
+++ b/configure.ac
@@ -1422,18 +1422,6 @@ if test x$use_boost = xyes; then
   AX_BOOST_SYSTEM
   AX_BOOST_FILESYSTEM
 
-  dnl Opt-in to Boost Process if external signer support is requested
-  if test "x$use_external_signer" != xno; then
-    AC_MSG_CHECKING(for Boost Process)
-    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <boost/process.hpp>]],
-     [[ boost::process::child* child = new boost::process::child; delete child; ]])],
-     [ AC_MSG_RESULT(yes)
-     AC_DEFINE([ENABLE_EXTERNAL_SIGNER],,[define if external signer support is enabled])
-     ],
-     [ AC_MSG_ERROR([Boost::Process is required for external signer support, but not available!])]
-    )
-  fi
-
   if test x$suppress_external_warnings != xno; then
     BOOST_CPPFLAGS=SUPPRESS_WARNINGS($BOOST_CPPFLAGS)
   fi
@@ -1441,6 +1429,9 @@ if test x$use_boost = xyes; then
   BOOST_LIBS="$BOOST_LDFLAGS $BOOST_SYSTEM_LIB $BOOST_FILESYSTEM_LIB"
 fi
 
+if test "x$use_external_signer" != xno; then
+  AC_DEFINE([ENABLE_EXTERNAL_SIGNER],,[define if external signer support is enabled])
+fi
 AM_CONDITIONAL([ENABLE_EXTERNAL_SIGNER], [test "x$use_external_signer" = "xyes"])
 
 dnl Check for reduced exports

--- a/doc/dependencies.md
+++ b/doc/dependencies.md
@@ -6,7 +6,7 @@ These are the dependencies currently used by Bitcoin Core. You can find instruct
 | Dependency | Version used | Minimum required | CVEs | Shared | [Bundled Qt library](https://doc.qt.io/qt-5/configure-options.html#third-party-libraries) |
 | --- | --- | --- | --- | --- | --- |
 | Berkeley DB | [4.8.30](https://www.oracle.com/technetwork/database/database-technologies/berkeleydb/downloads/index.html) | 4.8.x | No |  |  |
-| Boost | [1.71.0](https://www.boost.org/users/download/) | [1.58.0](https://github.com/bitcoin/bitcoin/pull/19667) | No |  |  |
+| Boost | [1.71.0](https://www.boost.org/users/download/) | [1.64.0](https://github.com/bitcoin/bitcoin/pull/22320) | No |  |  |
 | Clang |  | [5.0+](https://releases.llvm.org/download.html) (C++17 support) |  |  |  |
 | Expat | [2.2.7](https://libexpat.github.io/) |  | No | Yes |  |
 | fontconfig | [2.12.1](https://www.freedesktop.org/software/fontconfig/release/) |  | No | Yes |  |

